### PR TITLE
Spotlight: don't join a public room when clicking on it

### LINKED ISSUE #27523: No way to copy aliases/ids from the room directory without accidentally joining rooms
### Steps to reproduce

1. Search the room directory for something
2. Want to copy the alias/roomid that shows up
3. Try to select the text by clicking & dragging over it
4. Accidentally join the room

#### What did you expect?

Looks like we might be triggering joins by onMouseDown rather than onClick or something?  Or perhaps just require the user to explicitly click the Join or Preview button if they actually want to join the room?

#### What happened instead?

Accidentally joining rooms.

### Operating system

_No response_

### Application version

Element Nightly version: 2024060501 Crypto version: Rust SDK 0.7.0 (068a0af), Vodozemac 0.6.0

### How did you install the app?

_No response_

### Homeserver

_No response_

### Will you send logs?

No

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

Closes https://github.com/element-hq/element-web/issues/27523

For public room, the listener is only reacting to keyboard event for keyboard/accessibility navigation. The list element already has a button to join or preview a room for the mouse user.

### CODE REVIEW

File: src/components/views/dialogs/spotlight/SpotlightDialog.tsx [hunk 1]
```diff
@@ -710,7 +710,6 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
                         id={`mx_SpotlightDialog_button_result_${result.publicRoom.room_id}`}
                         className="mx_SpotlightDialog_result_multiline"
                         key={`${Section[result.section]}-${result.publicRoom.room_id}`}
-                        onClick={listener}
```
Comment (t3chguy) [inline src/components/views/dialogs/spotlight/SpotlightDialog.tsx:713]: Won't this break keyboard accessibility?

**Review by t3chguy — CHANGES_REQUESTED** [review]
This has a very bad UX right now. The hover effect + cursor pointer makes it look clickable but clicking on it does nothing, it feels broken.

**Comment by t3chguy** [comment]
Given this moves away from the designs provided to us I think we will want a design review to confirm they are happy with the new UX.

**Review by gaelledel — APPROVED** [review]
Thx - all good. To think about if we're willing to change the colour on Hover now or later.

**Comment by florianduros** [comment]
@gaelledel the colour change should be easy, will do in the PR

**Review by t3chguy — CHANGES_REQUESTED** [review]
Keyboard accessibility seems broken. Enter is not working to activate the tile


https://github.com/matrix-org/matrix-react-sdk/assets/2403652/44b900f2-326a-4f54-9c77-0f8a5e3c42fc




Otherwise this does what it says on the tin but I am worried about the inconsistency in the Spotlight dialog when you switch into `Public rooms/spaces` more. Prior the whole tile is clickable, after only the button is clickable. I feel like this inconsistency is not ideal for a11y.

**Pull request closed without merging**